halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/services/v1/GenerateService.java

```diff
@@ -18,30 +18,62 @@
 
 import com.netflix.spinnaker.halyard.config.config.v1.AtomicFileWriter;
 import com.netflix.spinnaker.halyard.config.errors.v1.HalconfigException;
+import com.netflix.spinnaker.halyard.config.model.v1.node.Halconfig;
 import com.netflix.spinnaker.halyard.config.model.v1.node.NodeFilter;
-import com.netflix.spinnaker.halyard.config.model.v1.problem.Problem;
 import com.netflix.spinnaker.halyard.config.model.v1.problem.ProblemBuilder;
 import com.netflix.spinnaker.halyard.config.spinnaker.v1.component.SpinnakerComponent;
-import java.io.IOException;
-import java.nio.file.FileSystem;
-import java.nio.file.FileSystems;
-import java.nio.file.Path;
-import java.util.ArrayList;
-import java.util.List;
 import lombok.extern.slf4j.Slf4j;
+import org.apache.tomcat.util.http.fileupload.FileUtils;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Component;
 
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.*;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import static com.netflix.spinnaker.halyard.config.model.v1.problem.Problem.Severity;
+
 @Component
 @Slf4j
 public class GenerateService {
   @Autowired
   String spinnakerOutputPath;
 
+  @Autowired
+  String halconfigDirectory;
+
+  @Autowired
+  String halconfigPath;
+
   @Autowired(required = false)
   List<SpinnakerComponent> spinnakerComponents = new ArrayList<>();
 
   public void generateConfig(NodeFilter nodeFilter) {
+    String deployment = nodeFilter.getDeployment();
+    if (deployment == null || deployment.isEmpty()) {
+      throw new HalconfigException(
+          new ProblemBuilder(Severity.FATAL, "No deployment to generate config for specified.").build());
+    }
+
+    log.info("Generating config from \"" + halconfigPath + "\" with deployment \"" + deployment + "\"");
+
+    File spinnakerOutput = new File(spinnakerOutputPath);
+
+    try {
+      FileUtils.deleteDirectory(spinnakerOutput);
+    } catch (IOException e) {
+      throw new HalconfigException(
+          new ProblemBuilder(Severity.FATAL, "Unable to clear old spinnaker config: " + e.getMessage() + ".").build());
+    }
+
+    if (!spinnakerOutput.mkdir()) {
```

Comment:
If you want to also create missing intermediate dirs, you can use `mkdirs()` here.